src/PrestaShopBundle/Controller/Admin/ProductController.php

```diff
@@ -1190,8 +1193,11 @@ public function toggleStatusAction($productId)
 
         try {
             $isEnabled = $this->getQueryBus()->handle(new GetProductIsEnabled($productId));
-
-            $this->getCommandBus()->handle(new UpdateProductStatusCommand($productId, !$isEnabled));
+            $this->getCommandBus()->handle(new UpdateProductStatusCommand(
```

Comment:
`GetProductIsEnabled` should also be fetched with a shop constraint:
- when you are in single shop context it will get the value for this shop and correctly invert it
- when you are in All shops context, the value returned should be the one from the product's default shop which is the one displayed in the list by default so it still makes sense to use it as a reference